src/vox_scene_info.rs

```diff
@@ -0,0 +1,213 @@
+use bevy::{
+    prelude::{
+        AssetServer, Assets, BuildChildren, Commands, Component, ComputedVisibility, Entity,
+        GlobalTransform, Handle, Mesh, Name, PbrBundle, StandardMaterial, Transform, Visibility,
+    },
+    reflect::{TypePath, TypeUuid},
+    utils::HashMap,
+};
+use dot_vox::{Layer, SceneNode};
+
+#[derive(Debug, TypeUuid, TypePath, Clone)]
+#[uuid = "39cadc56-aa9c-4543-8640-a018b74b5052"]
+
+pub struct VoxSceneInfo {
+    pub scenes: Vec<SceneNode>,
+    pub layers: Vec<Layer>,
+    // layers hidden status
+    pub layer_map: HashMap<u32, bool>,
+}
+
+#[derive(Debug, Clone, Component)]
+pub struct LayerData(pub u32);
+
+impl VoxSceneInfo {
+    pub fn new(scenes: Vec<SceneNode>, layers: Vec<Layer>) -> Self {
+        Self {
+            scenes: scenes,
+            layers: layers.clone(),
+            layer_map: layers
+                .clone()
+                .into_iter()
+                .map(|x| {
+                    let number = x
+                        .attributes
+                        .get("_name")
+                        .map(|s| s.parse().unwrap())
+                        .unwrap_or(u32::MAX);
+                    let hidden = x
+                        .attributes
+                        .get("_hidden")
+                        .map(|s| s == "1")
+                        .unwrap_or(false);
+                    (number, hidden)
+                })
+                .collect(),
+        }
+    }
+
+    pub fn all_loaded(
+        &self,
+        base_id: &'static str,
+        mesh_assets: &Assets<Mesh>,
+        asset_server: &AssetServer,
+    ) -> bool {
+        for node in self.scenes.iter() {
+            match node {
+                SceneNode::Shape {
+                    attributes: _,
+                    models,
+                } => {
+                    for shape in models {
+                        let key = if shape.model_id == 0 {
+                            format!("{}", base_id,)
+                        } else {
+                            format!("{}#model{}", base_id, shape.model_id)
+                        };
+                        let handle: Handle<Mesh> = asset_server.get_handle(key);
+                        if mesh_assets.get(&handle).is_none() {
+                            return false;
+                        }
+                    }
+                }
+                _ => {}
+            }
+        }
+        return true;
+    }
+
+    pub fn to_entity(
```

Comment:
I would rather have this called `spawn` or something along this at it effectively spawns an entity into the world.
```suggestion
    pub fn spawn(
```